### DISCUSSION TITLE
Encounter runtime_error during run_video_slam

### DISCUSSION THREAD

**Discussion by songwenhuang** [stella-cv/stella_vslam#75] [Q&A] [answered]
root@host:/openvslam/build# ./run_video_slam -v ../orb_vocab/orb_vocab.dbow2 -m ../aist_living_lab_1/video.mp4 -c ../aist_living_lab_1/config.yaml --frame-skip 3 --no-sleep --map-db map.msg

[2021-04-27 04:41:59.412] [I] loading ORB vocabulary: ../orb_vocab/orb_vocab.dbow2
terminate called after throwing an instance of 'std::runtime_error'
  what():  Vocabulary::fromStream invalid signature
*** Aborted at 1619498519 (unix time) try "date -d @1619498519" if you are using GNU date ***
PC: @     0x7fedf4c67438 gsignal
*** SIGABRT (@0x10) received by PID 16 (TID 0x7fedf83cf840) from PID 16; stack trace: ***
    @     0x7fedf55a5390 (unknown)
    @     0x7fedf4c67438 gsignal
    @     0x7fedf4c6903a abort
    @     0x7fedf52a184d __gnu_cxx::__verbose_terminate_handler()
    @     0x7fedf529f6b6 (unknown)
    @     0x7fedf529f701 std::terminate()
    @     0x7fedf529f919 __cxa_throw
    @     0x7fedf3aa686a fbow::Vocabulary::fromStream()
    @     0x7fedf3aa6e8b fbow::Vocabulary::readFromFile()
    @     0x7fedf7995951 openvslam::system::system()
    @           0x417aba mono_tracking()
    @           0x416521 main
    @     0x7fedf4c52840 __libc_start_main
    @           0x416a19 _start
    @                0x0 (unknown)
Aborted (core dumped)

Does anyone know how to fix this error? Thanks.

**Comment by dimaxano** [comment]
Does `../orb_vocab/orb_vocab.dbow2` really exists? Check if the path is right

**Reply by songwenhuang** [reply]
Yes. ../orb_vocab/orb_vocab.dbow2 does exist. Thanks for your advice.

**Comment by ymd-stella** [comment, marked as answer]
If you are using FBoW, use orb_vocab.fbow (in https://github.com/OpenVSLAM-Community/FBoW_orb_vocab) instead of orb_vocab.dbow2.

**Reply by songwenhuang** [reply]
@ymd-stella The issue was fixed after using orb_vocab.fbow instead. Thanks so much.
